src/Contracts/PhpSoap/Client.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace RicorocksDigitalAgency\Soap\Contracts\PhpSoap;
+
+use SoapHeader;
+
+interface Client
+{
+    public function setEndpoint(string $endpoint): static;
+
+    /**
+     * @param array<string, mixed> $options
+     */
+    public function setOptions(array $options): static;
+
+    /**
+     * Set the given headers on the request.
+     *
+     * @param array<int, SoapHeader> $headers
+     */
+    public function setHeaders(array $headers): static;
+
+    /**
+     * Make a request and return its response.
+     */
+    public function call(string $method, mixed $body): mixed;
+
+    /**
+     * Get an array of supported SOAP functions.
+     *
+     * @return array<int, string>
+     */
+    public function getFunctions(): array;
```

Comment:
Our php-soap engine returns more information about the functions than a regular string.
https://github.com/php-soap/engine/blob/main/src/Metadata/Metadata.php

We could parse them back to a regular int, string - but it means you will loose some information.
Alternatively we could add a generic for determing what kind of information this function will return?